binstar_client/scripts/cli.py

```diff
@@ -49,6 +56,121 @@ def file_or_token(value: str) -> str:
     return value
 
 
+def json_action(action):
+    a_data = dict(action._get_kwargs())
+
+    if a_data.get('help'):
+        a_data['help'] = a_data['help'] % a_data
+
+    if isinstance(action , argparse._SubParsersAction):
+        a_data.pop('choices', None)
+        choices = {}
+        for choice in action._get_subactions():
+            choices[choice.dest] = choice.help
+        a_data['choices'] = choices
+
+    reg = {v:k for k, v in action.container._registries['action'].items()}
+    a_data['action'] = reg.get(type(action), type(action).__name__)
+    if a_data['action'] == 'store' and not a_data.get('metavar'):
+        a_data['metavar'] = action.dest.upper()
+
+    a_data.pop('type', None)
+    a_data.pop('default', None)
+
+    return a_data
+
+
+def json_group(group):
+    grp_data = {'description': group.description,
+                'title': group.title,
+                'actions': [json_action(a) for a in group._group_actions if a.help != argparse.SUPPRESS],
+                }
+
+    if group._action_groups:
+        grp_data['groups'] = [json_group(g) for g in group._action_groups]
+
+    return grp_data
+
+
+class json_help(argparse.Action):
+    def __init__(self, nargs=0, help=argparse.SUPPRESS, **kwargs):
+        argparse.Action.__init__(self, nargs=nargs, help=help, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string):
+        self.nargs = 0
+        docs = {'prog': parser.prog,
```

Comment:
I'm not sure what the code formatting policy is for anaconda-client, but that looks weird.

Reply:
Yep, no idea either.
> \[...\] verbatim copy of that code.
> 
> N.B.: If code style, annotation, or other changes are desired, please feel free to push to this PR's branch \[...\]

Reply:
(And thanks for reviewing!)

Reply:
CI log indicated which linters are used.
Commits to make those happy have been added.